src/lua/LuaPlayer.cpp

```diff
@@ -310,5 +313,32 @@ bool LuaPlayer::IsAttackable(unsigned char otherPlayerId)
 
 void LuaPlayer::SuggestPact(unsigned char otherPlayerId, const PactType pt, const unsigned duration)
 {
-    player.SuggestPact(otherPlayerId, pt, duration);
+    if (!player.isHuman())
+    {
+        AIPlayer* ai = GAMESERVER.GetAIPlayer(player.GetPlayerId());
+        AIInterface aii = ai->getAIInterface();
+        aii.SuggestPact(otherPlayerId, pt, duration);
+    }
 }
+
+void LuaPlayer::AcceptPact(const PactType pt, unsigned char otherPlayerId)
+{
+    if (!player.isHuman())
+    {
+        AIPlayer* ai = GAMESERVER.GetAIPlayer(player.GetPlayerId());
+        AIInterface aii = ai->getAIInterface();
+        GameWorldGame& world = player.GetGameWorld();
+        aii.AcceptPact(world.GetEvMgr().GetCurrentGF(), pt, otherPlayerId);
+    }
+}
+
+void LuaPlayer::CancelPact(const PactType pt, unsigned char otherPlayerId) 
+{
+    if (!player.isHuman())
```

Comment:
Good idea to have this as an extra function. Then accompany this with `onPactStateChanged(bool true=Created, false=canceled, ...)` (or `onPactCreated`,  `onPactCanceled`)

Reply:
I am a bit unsure, do you mean/prefer a method on LuaPlayer (which Lua may subscribe if its possbile(??)) or a Callback-Handler in LuaInterfaceGame (like EventSuggestPact)?

Reply:
I meant 2 extra events for lua like EventSuggestPact. Then we have pact suggested, approved, denied and canceled covered with events.